src/components/userpanel/JumboStampSystem.tsx

```diff
@@ -0,0 +1,632 @@
+    content: "";
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    z-index: 0;
+    background-color: #ffc13a;
+    clip-path: ${(props: JumboStampSystemProps) =>
+      props.hide
+        ? "polygon(0% 0,85% 0%,100% 100%,100% 100%,0% 100%)"
+        : "polygon(0% 0,74% 0%,90% 100%,75% 100%,0% 100%)"};
+    transition: clip-path 0.25s ease;
+  }
+`;
+
+const ArrowDecoration = styled.img`
+  position: relative;
+  z-index: 2;
+  height: 100%;
+  transform: ${(props: JumboStampSystemProps) =>
+    props.active ? "translate(10%, 0)" : "translate(0px, 0)"};
+  transform: ${(props: JumboStampSystemProps) =>
+    props.hide ? "translate(5%) scale(1.6)" : ""};
+  transition: all ease 1.25s;
+  animation: ${(props: JumboStampSystemProps) =>
+    props.hide
+      ? css`
+          ${moveIn} 1.6s cubic-bezier(1,0,0,1)
+        `
+      : css`
+          ${moveInBack} 1.6s cubic-bezier(1,0,0,1)
+        `};
+`;
+
+const RewardsHeaderContainer = styled.div`
+  background-color: #8397cf;
+  height: 4.05vh;
+  clip-path: var(--notched-sm-tp);
+  display: flex;
+  justify-content: flex-end;
+  padding: 1rem;
+`;
+
+const RewardsButtonContainer = styled.div`
+  display: flex;
+  justify-content: flex-end;
+`;
+
+const RewardsButtonInner = styled.div`
+  padding: 1rem 1rem 1rem 0;
+  width: fit-content;
+`;
+
+const RewardsHeader = styled.p`
+  font-size: 2.15vh;
+  color: white;
+  text-overflow: clip;
+  white-space: nowrap;
+  font-family: video-cond, serif;
+  font-weight: 300;
+`;
+
+const RewardsDate = styled.span`
+  padding: 0rem 1rem 0rem 1rem;
+  font-size: 2.25rem;
+  color: #e0ce83;
+  letter-spacing: -1px;
+`;
+
+const MainContainer = styled.div`
+  /* display: none; */
+  animation: ${(props: JumboStampSystemProps) =>
+    props.active
+      ? css`
+          ${changeView} 1s cubic-bezier(1,0,1,-0.07)
+        `
+      : css``};
+  transition: visibility ease 2s;
+  animation-fill-mode: forwards;
+  visibility: ${(props: JumboStampSystemProps) =>
+    props.active ? "hidden" : "visible"};
+  /* display: ${(props: JumboStampSystemProps) =>
+    props.active ? "none" : ""}; */
+`;
+
+interface JumboStampSystemProps {
+  active?: boolean;
+  hide?: boolean;
+}
+
+const JumboStampSystem = () => {
+  const dispatch = useDispatch();
+  const showing = useSelector(selectShowingStamp);
+
+  const [hoverActive, setHoverActive] = useState(false);
+
+  const { account } = useEthers();
+
+  function getStamps(): number[] | undefined {
+    const ids = Array.from({ length: 12 }, (_, i) => i + 1);
+    const { value, error } =
+      useCall(
+        {
+          contract: ContractInstance,
+          method: "balanceOfBatch",
+          args: [ids.map(() => account), ids],
+        },
+        {
+          chainId: 137,
+        }
+      ) ?? {};
+    if (error) {
+      return undefined;
+    }
+    return value?.[0].map((result: BigNumber) => Number(result));
+  }
+
+  const stamps = getStamps();
+
+  useEffect(() => {
+    if (stamps) {
```

Comment:
https://www.youtube.com/watch?v=CFRhGnuXG-4

Reply:
i even watched this video when it was uploaded 😭 i am a fake dev fr

Reply:
```javascript
✔️ Done. 
Changes made throughout components:

// useEffect(() => {
//     if (stamps) {
//       for (let i = 0; i < Math.floor(stamps.length / 3); i++) {
//         stampIndividual[i].edition[0].collected = Boolean(stamps[i]);
//         stampIndividual[i].edition[1].collected = Boolean(stamps[i + 4]);
//         stampIndividual[i].edition[2].collected = Boolean(stamps[i + 8]);

//         stampIndividual[i].tier1 = Boolean(stamps[i]);
//         stampIndividual[i].tier2 = Boolean(stamps[i + 4]);
//         stampIndividual[i].tier3 = Boolean(stamps[i + 8]);
//       }
//     }
//   }, [stamps]); 

- changed to

const updateStampIndividual = (
  stamps: number[],
  stampIndividual: IndividualStampType[]
) => {
  const iterations = Math.floor(stamps.length / 3);
  for (let i = 0; i < iterations; i++) {
    stampIndividual[i].edition[0].collected = Boolean(stamps[i]);
    stampIndividual[i].edition[1].collected = Boolean(stamps[i + 4]);
    stampIndividual[i].edition[2].collected = Boolean(stamps[i + 8]);

    stampIndividual[i].tier1 = Boolean(stamps[i]);
    stampIndividual[i].tier2 = Boolean(stamps[i + 4]);
    stampIndividual[i].tier3 = Boolean(stamps[i + 8]);
  }
};

useEffect(() => {
  if (stamps) {
    updateStampIndividual(stamps, stampIndividual);
  }
}, [stamps]);